### ISSUE TITLE
NoMethodError: undefined method `kernel' for #<Chef::Node::Attribute:…>

### ISSUE THREAD

**Issue by FGS-BMarkham** [rosstimson/chef-terraform#30]
Hi!

I'm getting an error running our test suite when I try to use the default recipe of this cookbook.

```
================================================================================
Recipe Compile Error in /tmp/d20180416-18365-1ocz5t8/cookbooks/terraform/attributes/default.rb
================================================================================

NoMethodError
-------------
undefined method `kernel' for #<Chef::Node::Attribute:0x00000009528768>

Cookbook Trace:
---------------
  /tmp/d20180416-18365-1ocz5t8/cookbooks/terraform/attributes/default.rb:3:in `from_file'

Relevant File Content:
----------------------
/tmp/d20180416-18365-1ocz5t8/cookbooks/terraform/attributes/default.rb:

  1:  node.default['terraform']['url_base'] = 'https://dl.bintray.com/mitchellh/terraform'
  2:  node.default['terraform']['version'] = '0.5.1'
  3>> node.default['terraform']['arch'] = kernel['machine'] =~ /x86_64/ ? 'amd64' : '386'
  4:  
  5:  # Transform raw output of the bintray checksum list into a Hash[filename, checksum].
  6:  # https://dl.bintray.com/mitchellh/terraform/${VERSION}_SHA256SUMS?direct
  7:  node.default['terraform']['raw_checksums'] = <<-EOF
  8:    191a4150c86f9b12d922e3300a5dc884f3fcfcbf0158d3c3a9d2c82e8738037b  terraform_0.5.1_darwin_386.zip
  9:    5915d7668b07ea3770f1bc8126764f90723eade0245e0634af3b051ae2ceb7e5  terraform_0.5.1_darwin_amd64.zip
 10:    c60f23cafd309faddb592adf42a85a8b672f8d88c050a1cf8d8365c5939d0744  terraform_0.5.1_freebsd_386.zip
 11:    5c84d810a5e2d813ed299650cb1b3e91210a1d189a08e8c531d4cd9d5e948ed8  terraform_0.5.1_freebsd_amd64.zip
 12:    87e8c4d758be1f040ce902952ef7f30330c34efde798282b230e8f8ab05f0d9b  terraform_0.5.1_freebsd_arm.zip

System Info:
------------
chef_version=13.5.3
platform=ubuntu
platform_version=16.04
ruby=ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
program_name=/var/lib/gems/2.3.0/gems/rspec-core-3.7.0/exe/rspec
executable=/var/lib/gems/2.3.0/gems/rspec-core-3.7.0/exe/rspec
```

From best I can tell, perhaps `chef-terraform` is using old syntax that is incompatible with newer versions of Chef? It appears the `daptiv/seven_zip` people [encountered the same thing](https://github.com/daptiv/seven_zip/issues/21) a couple of years ago.

**Comment by Fitzsimmons** [comment]
This appears to be fixed in master, but it hasn't been released to the chef supermarket.

**Comment by haidangwa** [comment]
Yes, it's merged into master, but if I recall, the tests hadn't passed yet.

**Comment by haidangwa** [comment]
Fixed and published v2.0.0

#31
